.github/workflows/linkcheck.yaml

```diff
@@ -25,9 +25,11 @@ jobs:
       - uses: actions/setup-python@v4
         with:
           python-version: "3.11"
+          cache: pip
+          cache-dependency-path: pyproject.toml
```

Comment:
I'm not so keen on caching because it often complicates debugging, even when it wasn't causing an issue as I may end up needing to rule out caching was an issue.

If we reference a lock file, it makes more sense to me though, but here we reference versions that are overall not very tightly pinned, so we want to test new versions as they arrive and fail just at the time when they arrive - and not when the cache times out at unknown time.

Example: I see that a failure is showing up in builds after 20th some month, but the new version of a package i wonder introduced a regression was released 16th. Without cache, I could say "ah, it must not have been that package released 16th because the fails showed up 20th and not 19th"

Due to that, I'd like us to remove the cache when we install from something else than a .lock file pinning dependencies super tight.